ephemeris/shed_tools.py

```diff
@@ -657,7 +694,53 @@ def install_repositories(self):
                 t.get('changeset_revision', "")
             ) for t in self.errored_repositories])
         )
-        log.info("All repositories have been processed.")
+        log.info("All repositories have been installed.")
+        if self.test:
+            installed_tools = []
+            target_repos = self.installed_repositories
+            if self.test_existing:
+                target_repos += self.skipped_repositories
+            for installed_repository in target_repos:
+                repo_tools = tools_for_repository(self.gi, installed_repository)
+                installed_tools.extend(repo_tools)
+
+            test_results = []
+
+            for tool in installed_tools:
+                galaxy_interactor_kwds = {
+                    "galaxy_url": re.sub('/api', '', self.gi.url),
+                    "master_api_key": self.gi.key,
```

Comment:
Will self.gi.key always return the key? Also if initiated with email and password. I tested it and it seems to be the case. But is this certain?

Reply:
https://github.com/galaxyproject/bioblend/blob/master/bioblend/galaxyclient.py#L202

Yes - as long as ``login_required=True`` with ``get_galaxy_connection`` - which it is for this command. It'd be nice if galaxy-lib had a variant of the client code that consume and used a bioblend GalaxyInstance but it doesn't yet so I think this is okay for now? I will admit this is a hack though - let me know if you need me to push the hack down a layer into galaxy-lib.